tests/wpunit/APITest.php

```diff
@@ -678,208 +678,605 @@ public function testGetGrowthStatsWithEndDate()
 	}
 
 	/**
-	 * Test that the `get_subscription_forms()` function returns expected data.
+	 * Test that get_forms() returns the expected data.
 	 *
 	 * @since   1.0.0
+	 *
+	 * @return void
 	 */
-	public function testGetSubscriptionForms()
+	public function testGetForms()
 	{
-		$this->markTestIncomplete();
+		$result = $this->api->get_forms();
 
-		$result = $this->api->get_subscription_forms();
-		$this->assertNotInstanceOf(WP_Error::class, $result);
-		$this->assertIsArray($result);
-		$this->assertArrayHasKey('id', reset($result));
-		$this->assertArrayHasKey('form_id', reset($result));
+		// Assert forms and pagination exist.
+		$this->assertDataExists($result, 'forms');
+		$this->assertPaginationExists($result);
+
+		// Iterate through each form, confirming no landing pages were included.
+		foreach ($result['forms'] as $form) {
+			// Assert shape of object is valid.
+			$this->assertArrayHasKey('id', $form);
+			$this->assertArrayHasKey('name', $form);
+			$this->assertArrayHasKey('created_at', $form);
+			$this->assertArrayHasKey('type', $form);
+			$this->assertArrayHasKey('format', $form);
+			$this->assertArrayHasKey('embed_js', $form);
+			$this->assertArrayHasKey('embed_url', $form);
+			$this->assertArrayHasKey('archived', $form);
+
+			// Assert form is not a landing page i.e embed.
+			$this->assertEquals($form['type'], 'embed');
+
+			// Assert form is not archived.
+			$this->assertFalse($form['archived']);
+		}
 	}
 
 	/**
-	 * Test that the `get_subscription_forms()` function returns a blank array when no data
-	 * exists on the ConvertKit account.
+	 * Test that get_forms() returns the expected data when
+	 * the status is set to archived.
 	 *
-	 * @since   1.0.0
+	 * @since   2.0.0
+	 *
+	 * @return void
 	 */
-	public function testGetSubscriptionFormsNoData()
+	public function testGetFormsWithArchivedStatus()
 	{
-		$this->markTestIncomplete();
+		$result = $this->api->get_forms('archived');
 
-		$result = $this->api_no_data->get_subscription_forms();
-		$this->assertNotInstanceOf(WP_Error::class, $result);
-		$this->assertIsArray($result);
-		$this->assertCount(0, $result);
+		// Assert forms and pagination exist.
+		$this->assertDataExists($result, 'forms');
+		$this->assertPaginationExists($result);
+
+		// Iterate through each form, confirming no landing pages were included.
+		foreach ($result['forms'] as $form) {
+			// Assert shape of object is valid.
+			$this->assertArrayHasKey('id', $form);
+			$this->assertArrayHasKey('name', $form);
+			$this->assertArrayHasKey('created_at', $form);
+			$this->assertArrayHasKey('type', $form);
+			$this->assertArrayHasKey('format', $form);
+			$this->assertArrayHasKey('embed_js', $form);
+			$this->assertArrayHasKey('embed_url', $form);
+			$this->assertArrayHasKey('archived', $form);
+
+			// Assert form is not a landing page i.e embed.
+			$this->assertEquals($form['type'], 'embed');
+
+			// Assert form is not archived.
```

Comment:
This comment is the opposite of what's below.

Reply:
My bad. Thanks for spotting!